models/google_play__country_report.sql

```diff
@@ -0,0 +1,150 @@
+with installs as (
+
+    select *
+    from {{ var('stats_installs_country') }}
+
+), ratings as (
+
+    select *
+    from {{ var('stats_ratings_country') }}
+
+
+), store_performance as (
+
+    select *
+    from {{ var('stats_store_performance_country') }}
+
+), install_metrics as (
+
+    select
+        *,
+        -- take the difference later or nah?
+        sum(daily_device_installs) over (partition by country, package_name rows between unbounded preceding and current row) as total_device_installs,
+        sum(daily_device_uninstalls) over (partition by country, package_name rows between unbounded preceding and current row) as total_device_uninstalls
+
+    from installs 
+
+), store_performance_metrics as (
+
+    select
+        *,
+        sum(store_listing_acquisitions) over (partition by country_region, package_name rows between unbounded preceding and current row) as total_store_acquisitions,
+        sum(store_listing_visitors) over (partition by country_region, package_name rows between unbounded preceding and current row) as total_store_visitors
+
+    from store_performance
+
+), country_join as (
+
+    select 
+        coalesce(install_metrics.date_day, ratings.date_day, store_performance_metrics.date_day) as date_day,
+        coalesce(install_metrics.country, ratings.country, store_performance_metrics.country_region) as country,
+        coalesce(install_metrics.package_name, ratings.package_name, store_performance_metrics.package_name) as package_name,
```

Comment:
Are there any unintended consequences of so many coalesces?

Reply:
was wondering if some cross-db compatibility stuff would come up when we implement circle ci (like does one db need us to used nested coalesces?) 

but other than that not that i can think of...

Reply:
Great! We shall see what we encounter when the time comes then haha